forms-flow-bpm/src/main/java/org/camunda/bpm/extension/hooks/rest/impl/TaskRestResourceImpl.java

```diff
@@ -1,20 +1,100 @@
 package org.camunda.bpm.extension.hooks.rest.impl;
 
-import org.camunda.bpm.extension.hooks.rest.RestResource;
+import org.camunda.bpm.engine.rest.TaskRestService;
+import org.camunda.bpm.engine.rest.dto.CountResultDto;
+import org.camunda.bpm.engine.rest.dto.VariableValueDto;
+import org.camunda.bpm.engine.rest.dto.task.CompleteTaskDto;
+import org.camunda.bpm.engine.rest.dto.task.IdentityLinkDto;
+import org.camunda.bpm.engine.rest.dto.task.TaskDto;
+import org.camunda.bpm.engine.rest.dto.task.TaskQueryDto;
 import org.camunda.bpm.extension.hooks.rest.TaskRestResource;
-import org.camunda.bpm.extension.hooks.rest.dto.CountResultDto;
-import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.camunda.bpm.extension.hooks.rest.dto.UserIdDto;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import javax.ws.rs.core.Request;
+import javax.ws.rs.core.Response;
 import javax.ws.rs.core.UriInfo;
+import java.util.List;
+import java.util.Map;
 
-@RestController
-@RequestMapping(RestResource.BASE_PATH+"/v1"+ TaskRestResource.PATH)
-public class TaskRestResourceImpl implements TaskRestResource{
+public class TaskRestResourceImpl implements TaskRestResource {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TaskRestResourceImpl.class);
+
+    private final TaskRestService restService;
+
+    public TaskRestResourceImpl(TaskRestService taskRestService) {
+        restService = taskRestService;
+    }
+
+    @Override
+    public Object getTasks(Request request, UriInfo uriInfo, Integer firstResult, Integer maxResults) {
+        return restService.getTasks(request, uriInfo, firstResult, maxResults);
+    }
+
+    @Override
+    public List<TaskDto>  queryTasks(TaskQueryDto query, Integer firstResult, Integer maxResults) {
+       return restService.queryTasks(query, firstResult, maxResults);
+    }
+
+    @Override
+    public CountResultDto getTasksCount(UriInfo uriInfo) {
+        return restService.getTasksCount(uriInfo);
```

Comment:
our standard way of returning data is by using hateoas. Try to use hateoas everywhere